### PR TITLE
Misc fixes found by running mypy against Dropbox internal codebase "C".

### PR DESCRIPTION
**Pull request by gvanrossum**
This is a bit of a grab-bag and I need to verify these against more code, but to get started I need it to be a PR...

### CODE REVIEW

File: stdlib/3/builtins.pyi [hunk 1]
```diff
@@ -50,7 +50,7 @@ class object:
     def __sizeof__(self) -> int: ...
     def __reduce__(self) -> tuple: ...
     def __reduce_ex__(self, protocol: int) -> tuple: ...
-    def __dir__(self) -> List[str]: ...
+    def __dir__(self) -> _Collection[str]: ...
```
Comment (JelleZijlstra) [inline stdlib/3/builtins.pyi]: Should this just be `Iterable`? Empirically, `dir()` is happy with a generator.

Comment (gvanrossum) [reply stdlib/3/builtins.pyi]: Agreed, I was being too clever. Done.

**Comment by gvanrossum** [comment]
@JelleZijlstra If you think this is harmless can you approve? Then I can merge it myself after running some more internal tests.

**Comment by gvanrossum** [comment]
Actually it's okay if you merge too -- testing showed no new issues.

File: stdlib/2/os/__init__.pyi [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 from builtins import OSError as error
 from io import TextIOWrapper as _TextIOWrapper
+from posix import stat_result as stat_result  # TODO: use this, see https://github.com/python/mypy/issues/3078
```
Comment (JelleZijlstra) [inline stdlib/2/os/__init__.pyi]: In Python 3 stats_result only exists in `os` but not `posix`. I'll fix it later.

**Pull request merged**